dnf/cli/commands/history.py

```diff
@@ -186,10 +187,11 @@ def _hcmd_redo(self, extcmds):
                 pass
 
     def _hcmd_undo(self, extcmds):
-        try:
-            return self.base.history_undo_transaction(extcmds[0])
-        except dnf.exceptions.Error as err:
-            return 1, [str(err)]
+        old = self.base.history_get_transaction(extcmds)
+        if old is None:
+            return 1, ['Failed history undo']
+        merged_trans = dnf.db.history.MergedTransactionWrapper(old)
```

Comment:
Is the `MergedTransactionWrapper` good for anything here? I sort of fail to see its usefulness in general, but here it seems to be completely unnecessary?

Reply:
removed, simplified